### ISSUE TITLE
linuxPackages_zen.kernel: fails to build since update to 5.18.0-zen1

### ISSUE THREAD

**Issue by InternetUnexplorer** [NixOS/nixpkgs#175879]
The Zen kernel is failing to build for me since the update to 5.18, and so I thought I'd open an issue since I can't seem to find any existing info about it.

**Bisected to https://github.com/NixOS/nixpkgs/commit/208fd4f17325cf506eea151e3614ae2081fa8b6c.**

### Steps To Reproduce

```sh
$ nix build github:NixOS/nixpkgs/208fd4f17325cf506eea151e3614ae2081fa8b6c#linuxPackages_zen.kernel -L
```

### Build Log

```
../drivers/i2c/busses/i2c-nct6775.c: In function 'nct6775_access':
../drivers/i2c/busses/i2c-nct6775.c:221:39: error: this statement may fall through [-Werror=implicit-fallthrough=]
  221 |                         tmp_data.byte = data->byte;
      |                         ~~~~~~~~~~~~~~^~~~~~~~~~~~
../drivers/i2c/busses/i2c-nct6775.c:222:17: note: here
  222 |                 case I2C_SMBUS_BYTE:
      |                 ^~~~
cc1: all warnings being treated as errors
make[3]: *** [../scripts/Makefile.build:288: drivers/i2c/busses/i2c-nct6775.o] Error 1
make[2]: *** [../scripts/Makefile.build:550: drivers/i2c/busses] Error 2
make[1]: *** [../scripts/Makefile.build:550: drivers/i2c] Error 2
make[1]: *** Waiting for unfinished jobs....
```

### Notify maintainers

@Atemu @andresilva 

### Metadata

- system: `"x86_64-linux"`
 - host os: `Linux 5.17.7-zen1, NixOS, 22.05 (Quokka), 22.05.20220518.48037fd`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.8.1`
 - channels(root): `""`
 - channels(alex): `""`
 - nixpkgs: `/run/current-system/nixpkgs`

**Comment by InternetUnexplorer** [comment]
Unfortunately 5.18.1 doesn't seem to build either for me. :( Given that the file in question hasn't been touched in months, I'm wondering if this has something to do with the move to C11 for the 5.18 kernel. But that's just a guess.

I can think of two possible quick solutions:
- Include a patch that adds a `/* fall through */` comment
- Ignore that warning everywhere in the kernel

**Comment by trofi** [comment]
Staging recently got https://github.com/NixOS/nixpkgs/pull/175433 change to avoid Werror-style failures.

**Comment by InternetUnexplorer** [comment]
Given that that change will take a bit to get to master, do you think it would make sense to set `WERROR = no` for just the zen kernel in the meantime? Or even just revert https://github.com/NixOS/nixpkgs/commit/208fd4f17325cf506eea151e3614ae2081fa8b6c and move it to staging so that it gets merged when the `WERROR` change does?

I'm asking because—while it's not too much trouble to add patches to my personal NixOS configuration—right now the zen kernel is broken for everyone, and so it might be a good idea to get it back to a working state in the meantime.

I'm not sure how many people use the zen kernel though, it seems to be not very many, and so if it's better to just leave it as-is for a bit that's fine too; I don't really know how such things are usually handled in nixpkgs and am trying to learn. :)

**Comment by trofi** [comment]
I'd say setting `WERROR = no;` for zen kernel is perfectly fine. Should be a matter of adding it to https://github.com/NixOS/nixpkgs/blob/master/pkgs/os-specific/linux/kernel/linux-zen.nix#L27. Do you want to give it a try and propose a PR? AFAIU you already run that kernel version and have a chance to validate it if still boots.

You could also add a TODO to remove it once staging is merged.

**Comment by Atemu** [comment]
https://github.com/NixOS/nixpkgs/pull/175433 should've gone to master and/or be merged with the next kernel bump or something if the rebuilds were *that* critical which I don't think they are. We could cherry-pick it if we need it now.

**Comment by InternetUnexplorer** [comment]
I made [a commit](https://github.com/NixOS/nixpkgs/commit/c240539dbaeb605c089c1d4a599f4e3044f1137e), but it's late here so I don't have time to test it properly and am waiting to open a PR until I can do so.

I figured while I was at it I should also bump the version to 5.18.1-zen1, but in hindsight maybe I should wait until the mainline kernel is updated to 5.18.1 in nixpkgs?

Anyway, I'll test it when I have time tomorrow evening, make any necessary changes, and open a PR, unless we decide to cherry-pick #175433? That seems like a nicer solution to me personally, but I don't really have a strong preference.
